hil/ext/switches/common.py

```diff
@@ -53,3 +56,36 @@ def parse_vlans(raw_vlans):
             vlan_list.append(num_str)
 
     return vlan_list
+
+
+def modify_port_common(self, port, channel, new_network):
+    """This implements modify port method that non-console like drivers can use
+    """
+    if channel == 'vlan/native':
+        if new_network is None:
+            self._remove_native_vlan(port)
+            self._port_shutdown(port)
+        else:
+            self._set_native_vlan(port, new_network)
+    else:
+        match = re.match(r'vlan/(\d+)', channel)
+
+        if match is None:
+            raise SwitchError("Malformed channel: No VLAN ID found")
+
+        vlan_id = match.groups()[0]
+        # we already do these checks in the API, are we being extra careful
+        # by doing it here or is this redundant?
```

Comment:
I think it's definitely a good idea to do the extra sanity check, but given that it should never fail, I'd use an `assert`.